### ISSUE TITLE
Expose active state for `Bar`

### ISSUE THREAD

**Issue by mmcc** [recharts/recharts#699]
## Do you want to request a feature or report a bug?
Request a feature

## What is the current behavior?
Nothing currently exposes any indication of which `<Bar />` is currently active in terms of tooltip hover state.

The real world use case I'm trying to make work is an interface where there are two synced charts: a `LineChart` and a `BarChart` below. The goal being that we could do things like change fill values when one of the synced charts has an active tooltip. Only way I can think to achieve this currently would be to hack it together by abusing custom tooltips.

Does this make sense? Any chance there's just something I'm missing?

**Comment by xile611** [comment]
@mmcc  
When you a same props `syncId`  to a `LineChart` and a `BarChart`, the tooltips of these two charts can be synced.


```jsx
<LineChart  syncId="someChart" />
<BarChart  syncId="someChart" />
```

**Comment by mmcc** [comment]
@xile611 Thanks for the response! Yeah, I've got the charts synced and everything's working, so the issue isn't getting sync functionality working, it's getting access to the active label at the `Bar` component level.

Looks like things have been refactored a bit since I tried to dig into this a few days ago, but the tl;dr is I was hoping there was something I was missing about getting access to `activePayload` without setting up a `onMouseMove` handler. I ended up getting halfway to what I want by setting up the mouse event handlers in a stateful component that wraps `BarChart`, but that doesn't allow me to show an active state on the `Bar` component when the `LineChart` above has an active tooltip.

Setting up the mouse events on a higher level component that wraps both charts felt reasonable, but then you get in the catch-22 of setting state and causing a re-render that stops the tooltip itself from being rendered.

If this kind of functionality isn't anywhere on the roadmap just let me know, I can see what I can do about a PR in the next few weeks.
